src/StoreApi/Schemas/ProductSchema.php

```diff
@@ -424,6 +519,146 @@ protected function get_product_quantity_limit( \WC_Product $product ) {
+		return is_a( $attribute, '\WC_Product_Attribute' );
+	}
+
+	/**
+	 * Returns true if the given attribute is valid and used for variations.
+	 *
+	 * @param mixed $attribute Object or variable to check.
+	 * @return boolean
+	 */
+	protected function filter_variation_attribute( $attribute ) {
+		return $this->filter_valid_attribute( $attribute ) && $attribute->get_variation();
+	}
+
+	/**
+	 * Get variation IDs and attributes from the DB.
+	 *
+	 * @param \WC_Product $product Product instance.
+	 * @returns array
+	 */
+	protected function get_variations( \WC_Product $product ) {
+		if ( ! $product->is_type( 'variable' ) ) {
+			return [];
+		}
+		global $wpdb;
+
+		$variation_ids               = $product->get_visible_children();
+		$attributes                  = array_filter( $product->get_attributes(), [ $this, 'filter_variation_attribute' ] );
+		$default_variation_meta_data = array_reduce(
+			$attributes,
+			function( $defaults, $attribute ) {
+				$meta_key              = wc_variation_attribute_name( $attribute->get_name() );
+				$defaults[ $meta_key ] = [
+					'name'  => wc_attribute_label( $attribute->get_name(), $product ),
+					'value' => null,
+				];
+				return $defaults;
+			},
+			[]
+		);
+
+		// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared
+		$variation_meta_data = $wpdb->get_results(
+			"
+			SELECT post_id as variation_id, meta_key as attribute_key, meta_value as attribute_value
+			FROM {$wpdb->postmeta}
+			WHERE post_id IN (" . implode( ',', array_map( 'esc_sql', $variation_ids ) ) . ")
+			AND meta_key IN ('" . implode( "','", array_map( 'esc_sql', array_keys( $default_variation_meta_data ) ) ) . "')
+		"
+		);
+		// phpcs:enable
+
+		$attributes_by_variation = array_reduce(
+			$variation_meta_data,
+			function( $values, $data ) {
+				$values[ $data->variation_id ][ $data->attribute_key ] = $data->attribute_value;
+				return $values;
+			},
+			array_fill_keys( $variation_ids, [] )
+		);
+
+		$variations = [];
+
+		foreach ( $variation_ids as $variation_id ) {
+			$attribute_data = $default_variation_meta_data;
+
+			foreach ( $attributes_by_variation[ $variation_id ] as $meta_key => $meta_value ) {
+				if ( '' !== $meta_value ) {
+					$attribute_data[ $meta_key ]['value'] = $meta_value;
+				}
+			}
+
+			$variations[] = (object) [
+				'id'         => $variation_id,
+				'attributes' => array_values( $attribute_data ),
+			];
+		}
+
+		return $variations;
+	}
+
+	/**
+	 * Get list of product attributes and attribute terms.
+	 *
+	 * @param \WC_Product $product Product instance.
+	 * @return array
+	 */
+	protected function get_attributes( \WC_Product $product ) {
+		$attributes = array_filter( $product->get_attributes(), [ $this, 'filter_valid_attribute' ] );
+		$return     = [];
+
+		foreach ( $attributes as $attribute_slug => $attribute ) {
+			// Only visible and variation attributes will be exposed by this API.
+			if ( ! $attribute->get_visible() && ! $attribute->get_variation() ) {
```

Comment:
Should this be an _or_ instead of an _and_? If I'm understanding this clause correctly, currently it only skips attributes that are not visible **and** are not variation attributes. But we would like to skip all attribute that meet one of these two conditions, right?